### PR TITLE
Migrate to Swift 5

### PR DESCRIPTION
**Pull request by sssbohdan**
• Make code more modern-looking
• Fix formatting in a lot of places
• Remove unnecessary test extension

### CODE REVIEW

**Comment by sssbohdan** [comment]
@mwoollard Can you please review when you have time, thanks

**Comment by mwoollard** [comment]
@sssbohdan Thanks for taking the time to update this very old code! Swift 2 I think :-) I'll have a look over the w/e at the PR.

File: SwiftGooglePolyline/String+GooglePolyline.swift [hunk 1]
```diff
@@ -26,207 +26,163 @@
 
 import MapKit
 
-extension String {
+public extension String {
+  init<S: Sequence>(googlePolylineLocationCoordinateSequence sequence:S)
```
Comment (mwoollard) [inline SwiftGooglePolyline/String+GooglePolyline.swift:30]: Add space sequence:S

File: SwiftGooglePolyline/String+GooglePolyline.swift [hunk 2]
```diff
@@ -26,207 +26,163 @@
...
+  where S.Iterator.Element == CLLocationCoordinate2D {
+    var encoder = CoordinateEncoder()
+    self.init(sequence.reduce("") {
+      $0 + encoder.encode(coordinate: $1)
+    })
+  }
+  
+  init<S: Sequence>(googlePolylineMapPointSequence sequence: S)
+  where S.Iterator.Element == MKMapPoint {
+    self.init(googlePolylineLocationCoordinateSequence:sequence.map { $0.coordinate })
```
Comment (mwoollard) [inline SwiftGooglePolyline/String+GooglePolyline.swift:40]: Add space after :

File: SwiftGooglePolyline/String+GooglePolyline.swift [hunk 3]
```diff
@@ -26,207 +26,163 @@
...
+  func makeCoordinateSequenceFromGooglePolyline() throws -> AnySequence<CLLocationCoordinate2D> {
+    return AnySequence<CLLocationCoordinate2D>(try PolylineSequence(self))
+  }
+  
+  func makeCoordinateArrayFromGooglePolyline() throws -> [CLLocationCoordinate2D] {
+    return [CLLocationCoordinate2D](try self.makeCoordinateSequenceFromGooglePolyline())
+  }
+  
+  func makeMKPolylineFromGooglePolyline() throws -> MKPolyline {
+    return MKPolyline(sequence:try self.makeCoordinateSequenceFromGooglePolyline())
```
Comment (mwoollard) [inline SwiftGooglePolyline/String+GooglePolyline.swift:70]: Add space after :

File: SwiftGooglePolyline/MKPolyline+GooglePolyline.swift [hunk 4]
```diff
@@ -27,20 +27,18 @@
...
-    
-    public convenience init<S:SequenceType where S.Generator.Element == CLLocationCoordinate2D>(sequence:S) {
-        let array = Array<CLLocationCoordinate2D>(sequence)
-        self.init(coordinates: UnsafeMutablePointer(array), count: array.count)
-    }
+  public convenience init(encodedGooglePolyline: String) throws {
+    self.init(sequence: try encodedGooglePolyline.makeCoordinateSequenceFromGooglePolyline())
+  }
+  
+  public convenience init<S: Sequence>(sequence:S) where S.Iterator.Element == CLLocationCoordinate2D {
```
Comment (mwoollard) [inline SwiftGooglePolyline/MKPolyline+GooglePolyline.swift:34]: Add space sequence:S

**Review by mwoollard — CHANGES_REQUESTED** [review]
Hi - just a few formatting issues (spacing next to parameter name) - if you can update for those will approve and merge. I'll then add SPM support !

Thanks again for your time and interest.
Best regards
Mark

**Comment by sssbohdan** [comment]
> Hi - just a few formatting issues (spacing next to parameter name) - if you can update for those will approve and merge. I'll then add SPM support !
> 
> Thanks again for your time and interest. Best regards Mark

Thanks for the review and thanks for implementing the algorithm :)

**Pull request merged**